src/game/client/input/keyboard.cpp

```diff
@@ -134,17 +134,27 @@ void Keyboard::Create_Stream_Messages()
     if (g_theMessageStream != nullptr) {
         GameMessage *msg = nullptr;
 
-        for (int i = 0; m_keys[i].key != 0; ++i) {
-            if (m_keys[i].state & KEY_DOWN) {
+        KeyboardIO *ptr = Get_First_Key();
+        while (ptr->key != 0) {
```

Comment:
You should keep this as a for loop keyboard ptr is never used outside of this loop. I'd suggest changing the name to `keyIO` or something more descriptive.